fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1382,7 +1389,11 @@ ACTOR Future<Void> assignMutationsToStorageServers(CommitBatchContext* self) {
 		state VectorRef<Optional<MutationRef>>* encryptedMutations =
 		    &trs[self->transactionNum].transaction.encryptedMutations;
 		ASSERT(encryptedMutations->size() == 0 || encryptedMutations->size() == pMutations->size());
-		state int64_t tenantId = trs[self->transactionNum].tenantInfo.tenantId;
+		state int64_t encryptDomain = trs[self->transactionNum].tenantInfo.tenantId;
+		if (self->pProxyCommitData->encryptMode.mode == EncryptionAtRestMode::CLUSTER_AWARE &&
```

Comment:
Is this a known issue that is getting fixed here? Am trying to understand why was it not an issue earlier?

Reply:
Earlier in optional tenant mode some transactions would have a valid tenant id since they belong to a tenant. Now with the cluster mode encryption even in those cases we want to force everything to use the default domain. That's what we're changing here